src/analyses/constant_propagator.cpp

```diff
@@ -82,14 +106,18 @@ void constant_propagator_domaint::assign_rec(
       dest_values.set_to(s, tmp);
     }
     else
-      dest_values.set_to_top(s);
+    {
+      if(is_assignment)
+        dest_values.set_to_top(s);
+    }
   }
   else
   {
     // it's an assignment, but we don't really know what object is being written
     // to on the left-hand side - bail and set all values to top to be on the
     // safe side in terms of soundness
-    dest_values.set_to_top();
+    if(is_assignment)
```

Comment:
How about changing the above `else` to `else if(is_assignment)` and then even simplifying the comment a bit?